core/src/main/java/org/apache/calcite/rel/rules/FilterProjectTransposeRule.java

```diff
@@ -142,21 +137,24 @@ protected FilterProjectTransposeRule(
   public void onMatch(RelOptRuleCall call) {
     final Filter filter = call.rel(0);
     final Project project = call.rel(1);
+    final RelBuilder relBuilder = call.builder();
+    final RelNode newFilterRel = createNewFilterBelowProject(filter, project, relBuilder);
 
-    if (RexOver.containsOver(project.getProjects(), null)) {
```

Comment:
I have the impression that this refactoring makes the control optional while it shouldn't. Plus the comment is helpful so I wouldn't remove it.

Reply:
Agree, this change would break backwards compatibility for anyone who defined their own FilterProjectTransposeRule by using the constructor in line 104.

Reply:
This check could probably be moved to `matches(...)` method.

Reply:
Ok, I'll revert the change.